src/main/java/redis/clients/jedis/JedisClusterInfoCache.java

```diff
@@ -116,50 +116,81 @@ public void discoverClusterSlots(Jedis jedis) {
     }
   }
 
+  private void discoverClusterSlots(Jedis jedis) {
+    List<Object> slots = jedis.clusterSlots();
+    this.slots.clear();
+
+    for (Object slotInfoObj : slots) {
+      List<Object> slotInfo = (List<Object>) slotInfoObj;
+
+      if (slotInfo.size() <= MASTER_NODE_INDEX) {
```

Comment:
It's old code "if (slotInfo.size() <= 2) {", which is absolutely not related to PR. You can ask original author.